### ISSUE TITLE
Enable `@typescript-eslint/explicit-member-accessibility` linting rule

### ISSUE THREAD

**Issue by ClFeSc** [hpi-sam/digital-fuesim-manv#1121]
This rule enforces explicit accessibility annotations for class members. This rule was enabled but apparently non-functional prior to #1120. In #1120 it was disabled as several hundred occurrences of violations of this rule were found. We have decided that fixing this in #1120 would not be ideal and instead want to do it in a separate PR.

See [this discussion in #1120](https://github.com/hpi-sam/digital-fuesim-manv/pull/1120#discussion_r2410157037) for more context.

**Comment by lukasrad02** [comment]
I just noticed that actually, the rule wasn't enabled prior to #1120.

When adding ESLint to this repo, the rule was added to the config twice: Once set to `warn`, once set to `off`. The `warn` line was commented out. #591 seemingly enabled the rule by removing the comment, but it did not go into effect, since the rule was still set to `off` a couple of lines below. The migration tool correctly set the rule to off, since after parsing the full old config file, this is the correct value for the rule, but also added it to the config twice, which then let us discover the issue.

Due to the history, I expect that the rule should be enabled and it was just forgotten to remove the additional `off` from the config in #591, so I would keep this issue open to finally fix the rule.

**Comment by lukasrad02** [comment]
My plan is to gradually enable the rule (e.g. by module or subdirectory of a module) to be able to actually find a sensible access modifier for each class member and not just set every member to `public`.
